ztrouy/frontend-capstone-api
# Feature - Games Views

### Purpose
To enable Users to view Games and all of their relevant details, such as Genres, Platforms, and whether or not they own said Game

### Files Changed
- Created file, wrote serializers, and wrote all basic CRUD views for Games in new `games.py`
- Improted Games View Set to `__init__.py` package in the `/views` directory
- Added routing for the Games View Set in `urls.py`

### To Test
Fetch, setup by running `pipenv shell` and `./seed_database.sh` in the project root directory, run the project via the debugger, then confirm the following
- GET endpoint `/games` returns a list of all Games, with each Game including their Id, Name, URL for their Header Image, list of Genres, list of Platforms, and a calculated property specifying if the User owns the Game or not
- GET endpoint `/games/{id}` returns the appropriate Game if it exists, and includes all of the information from the `/games` endpoint
- POST endpoint `/games` creates a new Game and returns the created Game when provided with valid JSON
- PUT endpoint `/games/{id}` updates the appropriate Game if it exists when provided with valid JSON
- DELETE endpoint `/games/{id}` deletes the appropriate Game if it exists when provided with valid JSON

closes #8
